pkg/kubelet/eviction/eviction_manager.go

```diff
@@ -324,10 +324,21 @@ func (m *managerImpl) synchronize(diskInfoProvider DiskInfoProvider, podFunc Act
 
 	// rank the thresholds by eviction priority
 	sort.Sort(byEvictionPriority(thresholds))
-	thresholdToReclaim := thresholds[0]
-	resourceToReclaim, found := signalToResource[thresholdToReclaim.Signal]
-	if !found {
-		klog.V(3).Infof("eviction manager: threshold %s was crossed, but reclaim is not implemented for this threshold.", thresholdToReclaim.Signal)
+	var thresholdToReclaim evictionapi.Threshold
+	var resourceToReclaim v1.ResourceName
```

Comment:
good catch.  I think this would be easier to read if we moved it to a helper function

Reply:
Not sure if the helper function is more readable:
```
func (m *managerImpl) findThreshold() (evictionapi.Threshold, v1.ResourceName, bool) {
	var thresholdToReclaim evictionapi.Threshold
	var resourceToReclaim v1.ResourceName
	foundAny := false
	thresholds := m.config.Thresholds
	for i := range thresholds {
		thresholdToReclaim = thresholds[i]
		var found bool
		resourceToReclaim, found = signalToResource[thresholdToReclaim.Signal]
		if !found {
			klog.V(3).Infof("eviction manager: threshold %s was crossed, but reclaim is not implemented for this threshold.", thresholdToReclaim.Signal)
		} else {
			foundAny = true
			break
		}
	}
	return thresholdToReclaim, resourceToReclaim, foundAny
}
```

Reply:
I was thinking something like:
```golang
func getReclaimableThreshold(thresholds []evictionapi.Threshold) (evictionapi.Threshold, v1.ResourceName, bool) {
        sort.Sort(byEvictionPriority(thresholds))
	for i, threshold := range thresholds {
		if resource, ok := signalToResource[threshold.Signal]; ok {
                        return threshold, resource, true
		}
		klog.V(3).Infof("eviction manager: threshold %s was crossed, but reclaim is not implemented for this threshold.", threshold.Signal)
	}
	return nil, nil, false
}
```

It is easy to read, and should be easy to unit test without having to do a full eviction manager unit test.

Reply:
I got:

cannot use nil as type evictionapi.Threshold in return statement

Reply:
ah, yeah.  It would have to be
`return evictionapi.Threshold{}, v1.ResourceName{}, false`